db/rdbms/userFunctions.go

```diff
@@ -22,7 +22,7 @@ func (r *repository) GetUserById(uRef *models.User, id uint) error {
 	if id <= 0 {
 		return errors.New("user ID cannot be <= 0")
 	}
-	return r.DB.Find(uRef, id).Error
+	return r.DB.Take(uRef, id).Error
```

Comment:
`Find` doesn't produce an error if no records are found, while `Take` does. Using `Take` and checking for an error is how I determine if a user already exists [on line 208 of authentication.go](https://github.com/mike2ykme/FCC-Dallas-cohort-C/pull/28/files#diff-ef04a5b371555da95434437c8862f5628bc5d037338b63ebac9aa2a67517a6a6R208)

Reply:
Are we sure it doesn’t return an error? I thought there were cases during testing that we found it would return errors in prod DB setups.

Reply:
I couldn't get it to return an error when I tried. Going by the GORM documentation [here](https://gorm.io/docs/query.html), it seems like it is not supposed to:

> If you want to avoid the ErrRecordNotFound error, you could use Find like db.Limit(1).Find(&user), the Find method accepts both struct and slice data

I also tried checking the `Error` field of the returned struct when I was using `Find`, and that was `nil`. I wouldn't be shocked if I were missing something, however.